src/shell/shell_misc.cpp

```diff
@@ -50,6 +51,15 @@ static void outc(Bit8u c) {
 	DOS_WriteFile(STDOUT,&c,&n);
 }
 
+static void backone() {
+	BIOS_NCOLS;
+	uint8_t page = real_readb(BIOSMEM_SEG, BIOSMEM_CURRENT_PAGE);
+	if (CURSOR_POS_COL(page) > 0)
+		outc(8);
+	else if (CURSOR_POS_ROW(page) > 0)
+		INT10_SetCursorPos(CURSOR_POS_ROW(page) - 1, ncols - 1, page);
```

Comment:
We're doing the same calculation twice w/ `CURSOR_POS_ROW(page)`.
(not performance related), but we can make the code more DRY and make it a bit more self-documenting using a temporary variable.  Something like:
``` c++
	else if (const auto row = CURSOR_POS_ROW(page); row > 0) {
		BIOS_NCOLS;
		INT10_SetCursorPos(row  - 1, ncols - 1, page);
	}
```

That uses C++17 if with initializer:

```c++
if (init; condition)
```

Similar to what `for` has had for ages! 

It's nice because then that variable only exists inside the `if` block, which avoids a needing a variable before-hand that pollutes the outside scope.

Reply:
The if statement is enhanced in C++17, which is pretty useful indeed.